check_nextcloud.php

```diff
@@ -22,21 +23,75 @@ function convert_filesize($bytes, $decimals = 2) {
   return sprintf("%.{$decimals}f %s", $bytes / pow(1024, $factor), @$size[$factor]);
 }
 
+function check_range($pattern,$value) {
+    if(preg_match('/@([-0-9\.]+)\:([-0-9\.]+)/',$pattern,$matches)) {
+        return ( $value >= $matches[1] and $value <= $matches[2] )? 1 : 0;
+    }
+    elseif(preg_match('/([-0-9\.]+)\:([-0-9\.]+)/',$pattern,$matches)) {
+        return ( $value < $matches[1] or $value > $matches[2] )? 1 : 0;
+    }
+    elseif(preg_match('/\~\:([-0-9\.]+)/',$pattern,$matches)) {
+        return ( $value > $matches[1])? 1 : 0;
+    }
+    elseif(preg_match('/([-0-9\.]+)\:$/',$pattern,$matches)) {
+        return ( $value < $matches[1] ) ? 1 : 0;
+    }
+    elseif(preg_match('/([-0-9\.]+)/',$pattern,$matches)) {
+        return ($value < 0 or $value > $matches[1])?1:0;
+    }
+    else {
+        return 0;
+    }
+}       
+
 function show_helptext() {
   print "check_nextcloud.php - Monitoring plugin to check the status of nextcloud serverinfo app.\n
 You need to specify the following parameters:
-  -H:  hostname of the nextcloud instance, e.g. cloud.example.com
+  -H hostname: domain address of the nextcloud instance, e.g. cloud.example.com
   -U:  uri of the nextcloud serverinfo api, you'll find it at https://cloud.example.com/settings/admin/serverinfo
-  -T:  authenticate using serverinfo token (either -T or -u and -p)
+  -T token: authenticate using serverinfo token (either -T or -u and -p)
+  -P:  Performance Data Parameter:
+	freespace
+	load1
+	load5
+	load15
+	mem_free
+	mem_total
+	swap_free
+	swap_total
+	app_updates_available
+	users
+	users_active_5min
+	users_active_1h
+	users_active_24h
+	files
+	shares
+	shares_user
+	shares_groups
+	shares_link
+	shares_fed
+	php_version
+	db_size
+  -c range: Critical Value (returns 2 CRITICAL if -P out of range)
+  -w range:  Warning Value  (returns 1 WARNING if -P out of range)
```

Comment:
Also: Are the `-c <range>`  and `-w <range>` parameters used in conjunction with the `-P` parameter or are they standalone?

Reply:
ok